Il2CppInterop.Runtime/IL2CPP.cs

```diff
@@ -287,9 +287,20 @@ public static void ThrowIfNull(object arg)
             else
                 objectPointer = *(IntPtr*)objectPointer;
         }
-
         if (!valueTypeWouldBeBoxed && il2cpp_class_is_valuetype(Il2CppClassPointerStore<T>.NativeClassPtr))
-            objectPointer = il2cpp_value_box(Il2CppClassPointerStore<T>.NativeClassPtr, objectPointer);
+        {
+            uint align = 0;
+            if (typeof(T).IsValueType && il2cpp_class_value_size(Il2CppClassPointerStore<T>.NativeClassPtr, ref align) <= sizeof(IntPtr))
+            {
+                // NOTE(Kas): This assumes that objectPointer isn't pointing to the data, but is the actual data
+                // which has only started ocurring since adding support for generic byrefs in method proxygen
+                return Unsafe.As<IntPtr, T>(ref objectPointer);
+            }
+            else
+            {
+                objectPointer = il2cpp_value_box(Il2CppClassPointerStore<T>.NativeClassPtr, objectPointer);
+            }
+        }
```

Comment:
Is this change still necessary after 507a5ac30171b49df479e314c5ada772fee60f6b?

Reply:
Good catch, it's not necessary - I just got rid of it.